barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -292,11 +292,70 @@ void build_constraints(Builder& builder, AcirProgram& program, const ProgramMeta
+        if constexpr (IsUltraBuilder<Builder>) {
+            if (metadata.honk_recursion == 2) {
+                OpeningClaim<stdlib::grumpkin<Builder>> final_ipa_claim;
+                HonkProof final_ipa_proof;
+                if (honk_output.nested_ipa_claims.size() == 2) {
+                    // If we have two claims, accumulate.
+                    auto commitment_key = std::make_shared<CommitmentKey<curve::Grumpkin>>(1 << CONST_ECCVM_LOG_N);
+                    using StdlibTranscript = bb::stdlib::recursion::honk::UltraStdlibTranscript;
+
+                    auto ipa_transcript_1 = std::make_shared<StdlibTranscript>(honk_output.nested_ipa_proofs[0]);
+                    auto ipa_transcript_2 = std::make_shared<StdlibTranscript>(honk_output.nested_ipa_proofs[1]);
+                    auto [ipa_claim, ipa_proof] =
+                        IPA<stdlib::grumpkin<Builder>>::accumulate(commitment_key,
+                                                                   ipa_transcript_1,
+                                                                   honk_output.nested_ipa_claims[0],
+                                                                   ipa_transcript_2,
+                                                                   honk_output.nested_ipa_claims[1]);
+                    // If this is the root rollup, do full IPA verification
+                    if (honk_output.is_root_rollup) {
+                        auto verifier_commitment_key =
+                            std::make_shared<VerifierCommitmentKey<stdlib::grumpkin<Builder>>>(
+                                &builder,
+                                1 << CONST_ECCVM_LOG_N,
+                                std::make_shared<VerifierCommitmentKey<curve::Grumpkin>>(1 << CONST_ECCVM_LOG_N));
+                        // do full IPA verification
+                        auto accumulated_ipa_transcript =
+                            std::make_shared<StdlibTranscript>(convert_native_proof_to_stdlib(&builder, ipa_proof));
+                        IPA<stdlib::grumpkin<Builder>>::full_verify_recursive(
+                            verifier_commitment_key, ipa_claim, accumulated_ipa_transcript);
+                    } else {
+                        final_ipa_claim = ipa_claim;
+                        final_ipa_proof = ipa_proof;
+                    }
+                } else if (honk_output.nested_ipa_claims.size() == 1) {
+                    // If we have one claim, just forward it along.
+                    final_ipa_claim = honk_output.nested_ipa_claims[0];
+                    // This conversion looks suspicious but there's no need to make this an output of the circuit since
+                    // its a proof that will be checked anyway.
+                    final_ipa_proof = convert_stdlib_proof_to_native(honk_output.nested_ipa_proofs[0]);
+                } else if (honk_output.nested_ipa_claims.size() > 2) {
```

Comment:
Can you rearrange these conditions so that the default abort case is last? E.g. if 2, elif 1, elif 0, else throw